src/main/kotlin/org/rust/ide/utils/ExprUtils.kt

```diff
@@ -103,37 +103,46 @@ val RsUnaryExpr.operatorType: UnaryOperator?
 /**
  * Simplifies a boolean expression if can.
  *
+ * @param peek if true then does not perform any changes on PSI, 
+ *             `expr` is not defined and `result` indicates if this expression
+ *             can be simplified
  * @return `(expr, result)` where `expr` is a resulting expression,
  *         `result` is true if an expression was actually simplified.
  */
-fun RsExpr.simplifyBooleanExpression(): Pair<RsExpr, Boolean> {
+fun RsExpr.simplifyBooleanExpression(peek: Boolean = false): Pair<RsExpr, Boolean> {
```

Comment:
Cosmetics: function with a boolean argument whose value is a literal is usually a code smell. It's better to split this function in two with different names to enhance readability. 

In this case, I think we can do roughly like this:


```Kotlin
fn RsExpr.canBeSimplified(): Boolean {
    return simplifyBooleanExpression(dryRun=true).second
}

fn RsExpr.simplifyBooleanExpression(): RsExpr {
    return simplifyBooleanExpression(dryRun=false).first
}

// note that we hide ugly boolean argument API inside a private function
private fn RsExpr.simplifyBooleanExpression(dryRun: Boolean): Pair<RsExpr, Bool> {
    ....
}


```